warp10/src/main/java/io/warp10/script/functions/GEOCOVER.java

```diff
@@ -0,0 +1,120 @@
+
+public class GEOCOVER extends ElementOrListStackFunction {
+  
+  private final boolean rhumblines;
+  
+  public GEOCOVER(String name, boolean rhumblines) {
+    super(name);
+    this.rhumblines = rhumblines;
+  }
+  
+  @Override
+  public ElementStackFunction generateFunction(WarpScriptStack stack) throws WarpScriptException {
+    Object top = stack.pop();
+    
+    if (!(top instanceof Long)) {
+      throw new WarpScriptException(getName() + " expects a resolution (even number from 2 to 30).");
+    }
+    
+    final long resolution = ((Long) top).longValue();
+    
+    if (resolution < 2 || resolution > 30 || 1 == resolution % 2) {
+      throw new WarpScriptException(getName() + " expects a resolution (even number from 2 to 30).");
+    }
+    
+    final long mask = (0xFFFFFFFFFFFFFFFFL >>> (64 - (resolution * 2))) << (64 - (resolution * 2));
+    final long prefix = (resolution >>> 1) << 60;
+    
+    return new ElementStackFunction() {      
+      @Override
+      public Object applyOnElement(Object element) throws WarpScriptException {
+        
+        Set<Long> cells = new HashSet<Long>();
+
+        // If computing coverage for the rhumbline
+        if (rhumblines && element instanceof GTSEncoder) {
+          element = ((GTSEncoder) element).getDecoder().decode();
+        }
+        
+        if (element instanceof GeoTimeSerie) {
+          GeoTimeSerie gts = (GeoTimeSerie) element;
+          
+          if (rhumblines) {
+            GTSHelper.sort(gts);
+            List<Long> vertices = new ArrayList<Long>();
+            for (int i = 0; i < GTSHelper.nvalues(gts); i++) {
+              long location = GTSHelper.locationAtIndex(gts, i);
+              if (GeoTimeSerie.NO_LOCATION != location) {
+                vertices.add(location);
+              }
+            }            
+
+            for (long cell: HHCodeHelper.coverPolyline(vertices, (int) resolution, false).toGeoCells((int) resolution)) {
+              cells.add(cell);
+            }
+          } else {
+            for (int i = 0; i < GTSHelper.nvalues(gts); i++) {
+              long location = GTSHelper.locationAtIndex(gts, i);
+              if (GeoTimeSerie.NO_LOCATION != location) {
+                long cell = ((location & mask) >>> 4) | prefix;
+                cells.add(cell);
+              }
+            }            
+          }
+        } else if (element instanceof GTSEncoder) {          
+          if (rhumblines) {
+            throw new WarpScriptException(getName() + " hmm weird, we should never end here!");
+          } else {
+            GTSDecoder decoder = ((GTSEncoder) element).getDecoder();
+
+            while (decoder.next()) {
+              long location = decoder.getLocation();
+              if (GeoTimeSerie.NO_LOCATION != location) {
+                long cell = ((location & mask) >>> 4) | prefix;
+                cells.add(cell);
+              }
+            }            
+          }
+        }
+        
```

Comment:
ElementOrListStackFunction can work on any type so you still need to throw a WarpScriptException if the type of `element` is not what you expect.
So add a `else { throw ... }`.